magma/types/ready_valid.py

```diff
@@ -85,6 +85,14 @@ def fired(self):
         """
         raise NotImplementedError
 
+    @classmethod
```

Comment:
I think these are better off as free functions. We could then dispatch on both types and kinds (i.e. `is_producer(T()) \equiv is_producer(T)`)